x-pack/plugins/observability/public/pages/slo_edit/components/common/query_builder.tsx

```diff
@@ -76,11 +93,16 @@ export function QueryBuilder({
             isInvalid={fieldState.invalid}
             languageSwitcherPopoverAnchorPosition="rightDown"
             placeholder={placeholder}
-            query={{ query: String(field.value), language: 'kuery' }}
+            query={{ query: String(inputVal), language: 'kuery' }}
             size="s"
-            onChange={(value) => {
+            onSubmit={(value) => {
               field.onChange(value.query);
             }}
+            onChange={(value) => {
+              setInputVal(String(value.query));
+            }}
+            submitOnBlur={true}
+            validateKqlQuery={true}
```

Comment:
The component takes an `isInvalid` props as well as `validateKqlQuery`. 
It kind of gives two opposite meaning: The component is responsible for the validation, but the customer of the component is also responsible for the validation.
On top of that we duplicate the validation here and in the component.

What if we remove the validation from the component, and let the consumer validates and adds the error message below the input? 

```
render={({ field, fieldState }) => (
<>
      <QueryStringInput
            appName="Observability"
            bubbleSubmitEvent={false}
            dataTestSubj={dataTestSubj}
            disableAutoFocus
            disableLanguageSwitcher
            indexPatterns={dataView ? [dataView] : []}
            isDisabled={!dataView}
            isInvalid={fieldState.invalid}
            languageSwitcherPopoverAnchorPosition="rightDown"
            placeholder={placeholder}
            query={{ query: String(inputVal), language: 'kuery' }}
            size="s"
            onSubmit={(value) => {
              field.onChange(value.query);
            }}
            onChange={(value) => {
              setInputVal(String(value.query));
            }}
            submitOnBlur={true}
          />
        )}
      />
    {fieldState.invalid?.message ? <EuiText color="danger">{fieldState.invalid?.message}</EuiText> : null }
</>
```
Or we could provide a `errorText` or `helperText` prop that would take the error message and display it when not  undefined.

What do you think?

Reply:
@kdelemme i have applied suggested changes

Reply:
Thanks, testing the PR locally now